### PR TITLE
[Wallet][Refactoring] wallet/init refactoring backports

### PR DESCRIPTION
**Pull request by random-zebra**
Adapts the following refactoring PRs:

- bitcoin/bitcoin#6961 Constrain constant values to a single location in code
- bitcoin/bitcoin#7576 [Wallet] move wallet help string creation to CWallet
- bitcoin/bitcoin#7577 move "load wallet phase" to CWallet
- bitcoin/bitcoin#7608 Move hardcoded file name out of log messages
- bitcoin/bitcoin#7691 refactor wallet/init interaction
- bitcoin/bitcoin#8776 Wallet refactoring leading up to multiwallet

### CODE REVIEW

File: src/init.cpp [hunk 1]
```diff
@@ -457,7 +453,7 @@ std::string HelpMessage(HelpMessageMode mode)
 #ifdef ENABLE_WALLET
     strUsage += HelpMessageGroup(_("Wallet options:"));
     strUsage += HelpMessageOpt("-backuppath=<dir|file>", _("Specify custom backup path to add a copy of any wallet backup. If set as dir, every backup generates a timestamped file. If set as file, will rewrite to that file every backup."));
-    strUsage += HelpMessageOpt("-createwalletbackups=<n>", _("Number of automatic wallet backups (default: 10)"));
+    strUsage += HelpMessageOpt("-createwalletbackups=<n>", strprintf(_("Number of automatic wallet backups (default: 10)"), DEFAULT_CREATEWALLETBACKUPS));
```
[the remark this answers is not in the document]
Comment (random-zebra) [reply src/init.cpp]: Yep. It's fixed in the successive commit (when this option is moved to `CWallet::GetWalletHelpString`)

File: src/init.cpp [hunk 2]
```diff
@@ -462,7 +462,7 @@ std::string HelpMessage(HelpMessageMode mode)
     strUsage += HelpMessageOpt("-paytxfee=<amt>", strprintf(_("Fee (in %s/kB) to add to transactions you send (default: %s)"), CURRENCY_UNIT, FormatMoney(payTxFee.GetFeePerK())));
     strUsage += HelpMessageOpt("-rescan", _("Rescan the block chain for missing wallet transactions") + " " + _("on startup"));
     strUsage += HelpMessageOpt("-salvagewallet", _("Attempt to recover private keys from a corrupt wallet.dat") + " " + _("on startup"));
-    strUsage += HelpMessageOpt("-sendfreetransactions", strprintf(_("Send transactions as zero-fee transactions if possible (default: %u)"), fSendFreeTransactions));
+    strUsage += HelpMessageOpt("-sendfreetransactions", strprintf(_("Send transactions as zero-fee transactions if possible (default: %u)"), DEFAULT_SEND_FREE_TRANSACTIONS));
```
[the remark this answers is not in the document]
Comment (random-zebra) [reply src/init.cpp]: Sure, but out of scope of this PR.

[the remark this answers is not in the document]
Comment (furszy) [reply src/init.cpp]: yeah, just popped up during the review .

File: src/wallet/walletdb.cpp [hunk 3]
```diff
@@ -1013,6 +1013,7 @@ void ThreadFlushWalletDB(const std::string& strFile)
 
                 if (nRefCount == 0) {
                     boost::this_thread::interruption_point();
+                    const std::string& strFile = pwalletMain->strWalletFile;
```
[the remark this answers is not in the document]
Comment (random-zebra) [reply src/wallet/walletdb.cpp:1016]: The entire thread will be removed in bitcoin/bitcoin#9605

**Comment by random-zebra** [comment]
Rebased.

**Pull request merged**